include/compartment.h

```diff
@@ -29,11 +29,19 @@
 // about
 
 struct func_intercept;
+void compartment_transition_out();
+int64_t comp_exec_in(void*, void* __capability, void*);
+void comp_exec_out();
+
+#define INTERCEPT_INSTR_COUNT 5
+#define COMP_TRANS_FN_INSTR_CNT 4
 
 struct intercept_patch
 {
     int* patch_addr;
-    uint32_t instr;
+    int32_t instr[INTERCEPT_INSTR_COUNT];
+    uintptr_t comp_manager_cap_addr;
+    void* __capability manager_cap;
```

Comment:
I don't fully understand this struct, but I'm wondering if `manager_cap` can leak and thus allow a naughty compartment to upgrade its privileges?

Reply:
Theoretically no. This manager cap should only be a pointer to the two capabilities (`DDC`/`PCC` pair) required to perform the transition via `ldpblr`. However, I see that I derived it from the manager `DDC`, not from the compartment `DDC`, so I believe yes, it could be possible. I'll need to investigate further to be able to tell exactly.

Reply:
Ok.

Reply:
I've now sealed the capability in https://github.com/capablevms/CHERI-ELF-comp/pull/2/commits/446a6e381176542bd06ebb3980320078fefab2ff.